courses/views.py

```diff
@@ -379,11 +415,39 @@ def user_courses(request: HttpRequest) -> HttpResponse:
     template_name = "user/user_courses.html"
     context = {
         "user": request.user,
+        "token": _user_specific_token(request.user),
         "payment_account": settings.PAYMENT_ACCOUNT["default"],
     }
     return render(request, template_name, context)
 
 
+def _user_specific_token(user: User) -> str:
+    # without needing to extend the user, we can use its joined date (& time!),
+    # salt it with the secret key, and we should have an acceptable (-> hardly guessable) hash
+    str_to_hash = "{}-{}-{}".format(user.date_joined,
+                                    os.environ.get("SECRET_KEY"), user.username)
+    return hashlib.sha256(str_to_hash.encode()).hexdigest()
+
+
+def user_courses_ical(request: HttpRequest, user_id: int, security_token: str) -> HttpResponse:
+    user = get_object_or_404(User, pk=user_id)
+
+    security_token = request.GET.get('token', '')
+    if (security_token != _user_specific_token(user)):
+        raise PermissionDenied()
+
+    cal = Calendar()
+    cal.add('version', '2.0')
+    cal.add('prodid', "-//Tanzquotient user calendar {}//mxm.dk//".format(user_id))
+    subscriptions = user.profile.get_subscriptions()
```

Comment:
We need to filter for confirmed subscriptions. See `SubscribeState.ACCEPTED_STATES`